destination.go

```diff
@@ -112,20 +76,70 @@ func NewDestinationPlugin(impl Destination) cpluginv1.DestinationPlugin {
 }
 
 type destinationPluginAdapter struct {
-	impl       Destination
-	wgAckFuncs sync.WaitGroup
-	isAsync    bool
+	impl Destination
 
 	lastPosition *internal.AtomicValueWatcher[Position]
+	openCancel   context.CancelFunc
 
-	openCancel context.CancelFunc
+	// write is the chosen write strategy, either single records or batches
+	writeStrategy writeStrategy
 }
 
 func (a *destinationPluginAdapter) Configure(ctx context.Context, req cpluginv1.DestinationConfigureRequest) (cpluginv1.DestinationConfigureResponse, error) {
+	ctx = DestinationWithBatch{}.setBatchEnabled(ctx, false)
```

Comment:
Is this being set to false, because batching doesn't work yet? Or maybe due to the config checks?

Reply:
The way context stores values is by wrapping a context and returning a new context with that value. Like an onion. So essentially a context is immutable and can't be changed - if you send a context into a function it won't be able to add any values to _your specific context_, it can wrap it though and create a new context with a value.

So what we do here is add a value into the context, a pointer to be precise. The middleware then retrieves that pointer from the context and updates it, this way the change is visible to `destinationPluginAdapter`.